src/unit-number/unit-number-input.tsx

```diff
@@ -231,8 +246,8 @@ export const UnitNumberInput = ({
 
         if (onBlurRaw) {
             const valuesArr = [
-                floorValueStateRef.current,
-                unitValueStateRef.current,
+                formatInput(floorValueStateRef.current),
```

Comment:
move `formatInput` to the correct section. It should not be with the event handlers